packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewerHighlight/PdfViewerWithHighlight.tsx

```diff
@@ -13,35 +18,126 @@ type Props = PdfViewerProps & HighlightProps;
  */
 const PdfViewerWithHighlight: FC<Props> = ({
   highlightClassName,
+  activeHighlightClassName,
   document,
+  page,
   highlights,
+  activeIds,
   _useHtmlBbox,
   _usePdfTextItem,
+  setCurrentPage,
   ...rest
 }) => {
-  const { page, scale } = rest;
-  const [renderedText, setRenderedText] = useState<PdfRenderedText | null>(null);
+  const { scale } = rest;
+  const highlightProps: HighlightProps = {
+    highlightClassName,
+    activeHighlightClassName,
+    document,
+    highlights,
+    activeIds,
+    _useHtmlBbox,
+    _usePdfTextItem
+  };
 
+  const [renderedText, setRenderedText] = useState<PdfRenderedText | null>(null);
   const documentInfo = useAsyncFunctionCall(
     useCallback(async () => await extractDocumentInfo(document), [document])
   );
 
+  const activeHighlightPages = useActiveHighlightPages(document, highlights, activeIds);
+  const currentPage = useMovePageToActiveHighlight(page, activeHighlightPages, setCurrentPage);
+
   const highlightReady = !!documentInfo && !!renderedText;
   return (
-    <PdfViewer {...rest} setRenderedText={setRenderedText}>
+    <PdfViewer {...rest} page={currentPage} setRenderedText={setRenderedText}>
       <PdfViewerHighlight
-        highlightClassName={highlightClassName}
-        document={document}
         parsedDocument={highlightReady ? documentInfo ?? null : null}
         pdfRenderedText={highlightReady ? renderedText : null}
-        page={page}
-        highlights={highlights}
+        page={currentPage}
         scale={scale}
-        _useHtmlBbox={_useHtmlBbox}
-        _usePdfTextItem={_usePdfTextItem}
+        {...highlightProps}
       />
     </PdfViewer>
   );
 };
 
+/**
+ * Hook to move PDF page depending on active highlight
+ */
+function useMovePageToActiveHighlight(
+  page: number,
+  activeHighlightPages: (number | null)[],
+  setPage: ((page: number) => any) | undefined
+) {
+  const [highlightPage, setHighlightPage] = useState<number | undefined>();
+
+  useEffect(() => {
+    const pages = activeHighlightPages.filter(nonEmpty);
+    if (pages.length === 0 || pages.includes(page)) {
+      return;
+    }
+    const newPage = pages[0];
+    setHighlightPage(newPage);
+  }, [page, activeHighlightPages]);
+
+  const [currentPage, setCurrentPage] = useState(page);
```

Comment:
I think this function would read better if this were moved to the top and it referenced `currentPage` instead of `page` in the rest of the code.

Reply:
Moved the `currentPage useState` to the top, revised the code and added comments.